### DISCUSSION TITLE
How do I correctly override `Sequence.__getitem__`?

### DISCUSSION THREAD

**Discussion by Pyprohly** [microsoft/pyright#2953] [Q&A] [answered]
When I write:

```python
# pyright: strict

from typing import Sequence, Iterator, overload, Union

class MySequence(Sequence[float]):
    def __init__(self) -> None:
        self._seq: Sequence[float] = [1]

    def __len__(self) -> int:
        return len(self._seq)

    def __contains__(self, item: object) -> bool:
        return item in self._seq

    def __iter__(self) -> Iterator[float]:
        return iter(self._seq)

    @overload
    def __getitem__(self, index: int) -> float: ...
    @overload
    def __getitem__(self, index: slice) -> Sequence[float]: ...
    def __getitem__(self, index: Union[int, slice]) -> Union[float, Sequence[float]]:
        return self._seq[index]
```

It says:

```
error: Method "__getitem__" overrides class "Sequence" in an incompatible manner
  Return type mismatch: base method returns type "Sequence[float]", override returns type "float | Sequence[float]"
```

Are my type annotations for `def __getitem__` not right?

-----

In configuration.md, it says the default value for `reportMissingImports` is ’none’ but it is shown as ‘error’ in the Diagnostic Rule Defaults table.

**Comment by erictraut** [comment, marked as answer]
The check for override consistency isn't properly handling the case where the base class and the subclass both have overloads but the base class doesn't have a method implementation. I've changed the logic to detect this condition and skip the check to avoid this false positive error. This will be fixed in the next release.

I've fixed the documentation bug too. Thanks for noting this.

**Reply by Pyprohly** [reply]
Similar deal with `reportInvalidTypeVarUse` (‘none’ -> ‘warning’). The rest are all correct now.
